includes/class-sensei-analysis-overview-list-table.php

```diff
@@ -142,6 +149,141 @@ function get_columns() {
+	 * @since  4.3.0
+	 *
+	 * @param array $courses_ids Array of courses.
+	 * @return object[][] $lessons_grouped_by_courses Lessons array grouped by courses.
+	 */
+	private function get_lessons_by_courses( $courses_ids ) {
+		// Get Lessons by courses.
+		$lessons = Sensei()->course->get_lessons_grouped_by_courses( $courses_ids );
+
+		// Get Lessons grouped by courses.
+		$lessons_by_courses = array();
+		foreach ( $lessons as $lesson ) {
+			$lessons_by_courses[ $lesson->course_id ][] = $lesson->lesson_id;
+		}
+		return $lessons_by_courses;
+	}
+
+	/**
+	 * Get total average progress for all the courses.
+	 *
+	 * @since  4.3.0
+	 * @access private
+	 *
+	 * @param array $courses_ids List of courses ids.
+	 * @return string average_progress total average progress value and N/A if not applicable.
+	 */
+	private function get_total_average_progress_for_all_the_courses( $courses_ids ) {
+		$average_course_progress_total = 0;
+		if ( ! is_array( $courses_ids ) || empty( $courses_ids ) ) {
+			return $average_course_progress_total;
+		}
+		$students_by_courses = $this->get_students_by_courses( $courses_ids );
+		$lessons_by_courses  = $this->get_lessons_by_courses( $courses_ids );
+		foreach ( $courses_ids as $course_id ) {
+			// Calculate average progress per course.
+
+			// First step: Get total max value by multiplying students and lessons.
+			if ( ! is_array( $students_by_courses[ $course_id ] ) || ! is_array( $lessons_by_courses[ $course_id ] ) ) {
+				continue;
+			}
+			$lessons_students_max_value = ( count( $students_by_courses[ $course_id ] ) * count( $lessons_by_courses[ $course_id ] ) );
+
+			// Second step: Get completed lessons in the course.
+			$completed_lessons_in_course = Sensei()->lesson->get_completed_lessons_for_students_and_lessons( $lessons_by_courses[ $course_id ], $students_by_courses[ $course_id ] );
+			if ( $completed_lessons_in_course && is_array( $completed_lessons_in_course ) ) {
+				$average_course_progress_total += ( count( $completed_lessons_in_course ) / $lessons_students_max_value * 100 );
+			}
+		}
+		return $average_course_progress_total;
+	}
+
+	/**
+	 * Get all courses ids no pagination.
+	 *
+	 * @since  4.3.0
+	 *
+	 * @return array course ids array.
+	 */
+	private function get_all_courses_ids() {
+		// Get all courses ids.
+		$args = array(
+			'post_type'      => $this->post_type,
+			'post_status'    => array( 'publish', 'private' ),
+			'posts_per_page' => -1,
+			'fields'         => 'ID',
+		);
+
+		// Set search arguments.
+		if ( isset( $_GET['s'] ) && ! wp_verify_nonce( empty( $_GET['s'] ) ) ) {
+			$args['s'] = sanitize_text_field( wp_unslash( $_GET['s'] ) );
+		}
+
+		// Filter courses based on last activity filter if needed.
+		add_filter( 'posts_clauses', [ $this, 'filter_courses_by_last_activity' ] );
+		$courses_query = new WP_Query( $args );
+		remove_filter( 'posts_clauses', [ $this, 'filter_courses_by_last_activity' ] );
+
+		return array_column( $courses_query->posts, 'ID' );
+	}
```

Comment:
Do you think it would make more sense to not take into account the filtered courses so it will be more consistent with how all other totals are calculated? I think it would be awkward for one total to be affected by the filters and another not to. Maybe we should plan to fix it (if we can) for all the totals in the next release and not do it one by one in between releases?